context-free.md

```diff
@@ -1,22 +1,56 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+The language is targeted toward anyone with access to a computer (Windows,
+Mac, or Linux).  
 
 ## What is easy to do in this language? Why is it easy?
-
+It is easy to create shapes and modify their attributes.  This is because
+it provided built in functions for many primitive shapes (such as circles,
+triangles, squares, etc.) and color support.  It is also easy to compose
+shapes because the grammer allows for the definition of shapes to include
+shapes.
 
 ## What is hard to do in this language? Why is it hard?
-
+I found it difficult to learn the semantic model of the DSL.
+When I initially wanted to use recursion I was uncertain how to
+implement it since I did not find an obvious way to check for a base
+case or pass arguments to a function.  (Although there is explanation
+of pasing parameters to shapes in Version 3, it seems that no such
+feature exists in Version 2.)  Eventually I found (not in the documentation)
+but in one of the lesson files, `lesson.cfdg`, the following: 
+> A shape can use itself so long as it keeps getting smaller.
+> The system will stop the recursion when the shapes get too small to see.
+The fact that shapes were drawn from the center rather than one endpoint (or
+giving the choice of where to draw from) made it difficult in my case for
+me to determine without a bit of trial and error exactly how much I wanted to
+shift the x and y coordinates.
```

Comment:
The lack of precise documentation is one of the biggest drawbacks of CF. Way to overcome it!